components/pedigree/resources/src/main/resources/pedigree/model/dynamicGraph.js

```diff
@@ -280,18 +280,35 @@ define([
             } else {
                 // else: keep properties as defined in the pedigree (e.g. via one of the old importers)
                 //       and replace null RAW properties with an empty object
-                // TODO: once converters ar eupdated to fill new format fields instead of internal fields,
-                //       setting RAW properties t onull will not be neded nay more
+                // TODO: once converters are updated to fill new format fields instead of internal fields,
+                //       setting RAW properties to null will not be needed any more
                 this.setRawJSONProperties(id, {});
             }
         },
 
+        _isDisplayedAsConsanguinous: function( v ) {
+            var consangr = editor.getGraph().isConsangrRelationship(v);
+            var nodeConsangrPreference = editor.getView().getNode(v).getConsanguinity();
+            if (nodeConsangrPreference == "N")
+                consangr = false;
+            if (nodeConsangrPreference == "Y")
+                consangr = true;
+            return consangr;
+        },
+
         getPatientPhenotipsJSON: function( v )
         {
             if (!this.isPerson(v)) {
                 throw "Assertion failed: getPatientPhenotipsJSON() is applied to a non-person";
             }
-            return PhenotipsJSON.internalToPhenotipsJSON(this.getProperties(v), this.getRawJSONProperties(v));
+
+            var consangr = "N";
+            var parentPartnershipNode = editor.getGraph().getParentRelationship(v);
+            if (parentPartnershipNode && parentPartnershipNode !== null && this._isDisplayedAsConsanguinous(parentPartnershipNode)) {
+                consangr = "Y";
+            }
+
+            return PhenotipsJSON.internalToPhenotipsJSON(this.getProperties(v), this.getRawJSONProperties(v), consangr);
```

Comment:
I don't like that consanguinity is passed as a parameter, it's too custom. Can it instead be computed as part of `getProperties`?

Reply:
The issue here is that this is not a property of the node/person, but of the relationship and/or the pedigree layout (depending on how to look at this). So we may want to have a more generic parameter than consang., but it shouldn't be in the properties imho, maybe `relationshipProperties`?

Reply:
@allasm is the last commit along the lines of your proposed change?

Reply:
Yes

Reply:
It kind of is and isn't. As Brittney said, we can't really say that a person is consanguineous, but we do have a field in the patient record for marking that the person is descending from a consanguineous relationship, so that is a property of the patient. Since the information is found in the patient record (and the patient JSON), I think it should be part of `getProperties`.

On the other hand, `relationshipProperties` would be a good place to also store the parents' ethnicities.

Reply:
Pedigree and PT have different data models (where neither is "better", just one is centered around a single patient, another around the pedigree graph), in pedigree consanguinity is (in most cases) derived from pedigree structure, possibly (only) because of connections far away from either the person or the person-producing relationship. So in the context of pedigree, it does not make sense to have it in person properties - that is, until the moment we are forced to put it into PT-compatible JSON.

So having something like `relationshipProperties` may be the compromise solution in this case - in the future it can also be also useful to deal with parent ethnicities (as Sergiu suggested) which are not attached to a specific parent (as is the case for PT patients)